v1/sensor5200.js

```diff
@@ -0,0 +1,172 @@
+        const buffer = Buffer.from(response, 'base64');
+        const decodedResource = decodeResource(buffer, {
+          identifier: Number(resourcePath.split('/')[3]),
+          type: resourceType,
+        });
+        const resourceValue = decodedResource.value;
+        msg.payload = {
+          state: node.state,
+          data: {},
+        };
+        msg.payload.data[resourceName] = resourceValue;
+        node.cache[resourceName] = resourceValue;
+        msg.payload.cache = node.cache;
+        node.send(msg);
+      }).catch((err) => {
+        if (typeof err === 'number') {
+          node.error(`Error code: ${err}`);
+        } else {
+          node.error(err);
+        }
+      });
+    }
+
+    function configure() {
+      addResource('screenText', '/3341/0/5527', RESOURCE_TYPE.STRING, config.screenText);
+      addResource('temperatureBME', '/3303/2/5700', RESOURCE_TYPE.FLOAT, config.temperatureBME);
+      addResource('humidityBME', '/3304/2/5700', RESOURCE_TYPE.FLOAT, config.humidityBME);
+      addResource('pressureBME', '/3315/2/5700', RESOURCE_TYPE.FLOAT, config.pressureBME);
+      addResource('gasBME', '/3327/2/5700', RESOURCE_TYPE.FLOAT, config.gasBME);
+      addResource('accelerometerX', '/3313/0/5702', RESOURCE_TYPE.FLOAT, config.accelerometerX);
+      addResource('accelerometerY', '/3313/0/5703', RESOURCE_TYPE.FLOAT, config.accelerometerY);
+      addResource('accelerometerZ', '/3313/0/5704', RESOURCE_TYPE.FLOAT, config.accelerometerZ);
+      addResource('gyroscopeX', '/3334/0/5702', RESOURCE_TYPE.FLOAT, config.gyroscopeX);
+      addResource('gyroscopeY', '/3334/0/5703', RESOURCE_TYPE.FLOAT, config.gyroscopeY);
+      addResource('gyroscopeZ', '/3334/0/5704', RESOURCE_TYPE.FLOAT, config.gyroscopeZ);
+      addResource('magnetometerX', '/3314/0/5702', RESOURCE_TYPE.FLOAT, config.magnetometerX);
+      addResource('magnetometerY', '/3314/0/5703', RESOURCE_TYPE.FLOAT, config.magnetometerY);
+      addResource('magnetometerZ', '/3314/0/5704', RESOURCE_TYPE.FLOAT, config.magnetometerZ);
+      addResource('temperatureHDC', '/3303/1/5700', RESOURCE_TYPE.FLOAT, config.temperatureHDC);
+      addResource('humidityHDC', '/3304/1/5700', RESOURCE_TYPE.FLOAT, config.humidityHDC);
+      addResource('illuminance', '/3301/0/5700', RESOURCE_TYPE.FLOAT, config.illuminance);
+
+      node.device.write('/1/0/3', () => {
+      }, encodeResource({
+        identifier: 3,
+        type: RESOURCE_TYPE.INTEGER,
+        value: node.observationInterval,
+      }));
+
+      node.resources.forEach((resource) => {
+        if (resource.need) {
+          observe(resource.path, resource.name, resource.type);
+        }
+      });
+    }
+
+    node.on('input', (msg) => {
+      if (typeof msg.payload === 'string' || typeof msg.payload === 'number') {
+        let argument;
+        if (typeof msg.payload === 'string') {
+          argument = msg.payload;
+        } else if (typeof msg.payload === 'number') {
```

Comment:
Dabar šis kodas teisingas, bet ateityje keičiant 91 eilutę gali būti pamiršta atnaujinti susieti if'ai. Tokiais atvejais reikia arba argument inicializuoti į kokią nors default reikšmę, arba naudoti tiesiog `else` vietoj `else if`. Kadangi string objectas turi `toString` funkciją, šiuo atveju yra netgi paprastesnis sprendimas:

```javascript
let argument = msg.payload.toString();
node.device.write(...)
```